test/k8sT/Services.go

```diff
@@ -412,6 +412,158 @@ var _ = Describe("K8sServicesTest", func() {
+		)
+
+		var (
+			deploymentYAML string
+			lrpSvcYAML     string
+			ctx            context.Context
+			ctxCancel      context.CancelFunc
+			svcIP          string
+			curlTCP        string
+			curlUDP        string
+		)
+
+		BeforeAll(func() {
+			deploymentYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-test.yaml")
+			lrpSvcYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-svc.yaml")
+			res := kubectl.ApplyDefault(deploymentYAML)
+			res.ExpectSuccess("Unable to apply %s", deploymentYAML)
+			for _, pod := range []string{feFilter, beFilter} {
+				err := kubectl.WaitforPods(helpers.DefaultNamespace, fmt.Sprintf("-l %s", pod), helpers.HelperTimeout)
+				Expect(err).Should(BeNil())
+			}
+			ctx, ctxCancel = context.WithTimeout(context.Background(), 1*time.Second)
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, lrpServiceName)
+			svcIP = clusterIP
+			Expect(err).To(BeNil(), "Cannot get svc IP")
+			httpSVCURL := fmt.Sprintf("http://%s/", svcIP)
+			tftpSVCURL := fmt.Sprintf("tftp://%s/hello", svcIP)
+			curlTCP = helpers.CurlFail(httpSVCURL)
+			curlUDP = helpers.CurlFail(tftpSVCURL)
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(deploymentYAML)
+			ctxCancel()
+		})
+
+		It("LRP connectivity", func() {
+			// Basic sanity check
+			ciliumPods, err := kubectl.GetCiliumPods()
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+			for _, pod := range ciliumPods {
+				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, fmt.Sprintf("cilium service list | grep %s", svcIP), "Cannot retrieve services on cilium pod")
+				service.ExpectContains("LocalRedirect", "LocalRedirect is not present in the cilium service list")
+			}
+
+			By("Checking traffic goes to local backend")
+			testCases := []struct {
+				selector string
+				cmd      string
+				pod      string
+				want     string
+				notWant  string
+			}{
+				{
+					selector: "id=app1",
+					cmd:      curlTCP,
+					pod:      fe1Name,
+					// Expects to see local backend name in returned Hostname field
+					want: be1Name,
+					// Expects never to see remote backend name in returned Hostname field
+					notWant: be2Name,
+				},
+				{
+					selector: "id=app2",
+					cmd:      curlTCP,
+					pod:      fe2Name,
+					want:     be2Name,
+					notWant:  be1Name,
+				},
+				{
+					selector: "id=app1",
+					cmd:      curlUDP,
+					pod:      fe1Name,
+					want:     be1Name,
+					notWant:  be2Name,
+				},
+				{
+					selector: "id=app2",
+					cmd:      curlUDP,
+					pod:      fe2Name,
+					want:     be2Name,
+					notWant:  be1Name,
+				},
+			}
+			for _, tc := range testCases {
+				Consistently(func() bool {
+					res, err := kubectl.ExecInPods(ctx, helpers.DefaultNamespace, tc.selector, tc.cmd)
+					Expect(err).To(BeNil(), "%s failed in %s pod", tc.cmd, tc.selector)
+					returned := res[tc.pod].Stdout()
+					return strings.Contains(returned, tc.want) && strings.Contains(returned, tc.notWant)
```

Comment:
I think you meant this?
```suggestion
					return strings.Contains(returned, tc.want) && !strings.Contains(returned, tc.notWant)
```

Reply:
urghhh, fat fingers :) Fixed.